### ISSUE TITLE
Uninstallation via CLI is broken

### ISSUE THREAD

**Issue by tmoschou** [docker/for-mac#6598]
The uninstallation command `/Applications/Docker.app/Contents/MacOS/Docker --uninstall` as [documented](https://docs.docker.com/desktop/uninstall/) appears to be broken. Instead the command launches `Docker.app`, `Docker Desktop.app`, `com.docker.backend`, etc in the foreground and blocks. 

This is reproducible on all tested macOS workstations (including M1 and Apple Silicon).

Note that uninstalling via the graphical user interface is fine. However we are interested in scripting the uninstallation.

  - [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID: `044ED717-9763-4179-A514-EE5E0C93FED7/20221128070148`

### Expected behavior

```console
$ /Applications/Docker.app/Contents/MacOS/Docker --uninstall
Docker is running, exiting...
Docker uninstalled successfully. You can move the Docker application to the trash.
```

### Actual behavior

```console
$ /Applications/Docker.app/Contents/MacOS/Docker --uninstall
# Docker.app, Docker Desktop.app, com.docker.backend, etc launch in foreground...
# command does not return...
```

### Information

  - macOS Version: 13.0.1
  - Intel chip or Apple chip: Intel
  - Docker Desktop Version: 4.14.1 (91661)

### Output of `/Applications/Docker.app/Contents/MacOS/com.docker.diagnose check`

```
$ /Applications/Docker.app/Contents/MacOS/com.docker.diagnose check

Starting diagnostics

[PASS] DD0027: is there available disk space on the host?
[PASS] DD0028: is there available VM disk space?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0013: is the $PATH ok?
[PASS] DD0003: is the Docker CLI working?
[PASS] DD0014: are the backend processes running?
[PASS] DD0007: is the backend responding?
[PASS] DD0008: is the native API responding?
[PASS] DD0009: is the vpnkit API responding?
[PASS] DD0010: is the Docker API proxy responding?
[FAIL] DD0012: is the VM networking working? network checks failed: failed to ping host: exit status 1
[2022-11-28T07:07:47.168015000Z][com.docker.diagnose][I] ipc.NewClient: 49d63827-diagnose-network -> diagnosticd.sock diagnosticsd
[common/pkg/diagkit/gather/diagnose.runIsVMNetworkingOK()
[	common/pkg/diagkit/gather/diagnose/network.go:34 +0xd9
[common/pkg/diagkit/gather/diagnose.(*test).GetResult(0x100d57c00)
[	common/pkg/diagkit/gather/diagnose/test.go:46 +0x43
[common/pkg/diagkit/gather/diagnose.Run.func1(0x100d57c00)
[	common/pkg/diagkit/gather/diagnose/run.go:17 +0x5a
[common/pkg/diagkit/gather/diagnose.walkOnce.func1(0x2?, 0x100d57c00)
[	common/pkg/diagkit/gather/diagnose/run.go:142 +0x77
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x1, 0x100d57c00, 0xc000565720)
[	common/pkg/diagkit/gather/diagnose/run.go:151 +0x87
[common/pkg/diagkit/gather/diagnose.walkDepthFirst(0x0, 0x100d57d80, 0xc000565720)
[	common/pkg/diagkit/gather/diagnose/run.go:148 +0x52
[common/pkg/diagkit/gather/diagnose.walkOnce(0x100709ee0?, 0xc00059f888)
[	common/pkg/diagkit/gather/diagnose/run.go:137 +0xcc
[common/pkg/diagkit/gather/diagnose.Run(0x100d57d80, 0x10?, {0xc00059fb18, 0x1, 0x1})
[	common/pkg/diagkit/gather/diagnose/run.go:16 +0x1d4
[main.checkCmd({0xc0001aa010?, 0x6?, 0x4?}, {0x0, 0x0})
[	common/cmd/com.docker.diagnose/main.go:133 +0x105
[main.main()
[	common/cmd/com.docker.diagnose/main.go:99 +0x2a7
[2022-11-28T07:07:47.168857000Z][com.docker.diagnose][I] (4655ed64) 49d63827-diagnose-network C->S diagnosticsd POST /check-network-connectivity: {"ips":["10.109.0.71","10.64.64.14"]}
[2022-11-28T07:07:47.685305000Z][com.docker.diagnose][W] (4655ed64) 49d63827-diagnose-network C<-S d7bdfbee-diagnosticsd POST /check-network-connectivity (516.585273ms): failed to ping host: exit status 1

[SKIP] DD0030: is the image access management authorized?
[PASS] DD0019: is the com.docker.vmnetd process responding?
[PASS] DD0033: does the host have Internet access?
[PASS] DD0018: does the host support virtualization?
[PASS] DD0001: is the application running?
[PASS] DD0017: can a VM be started?
[PASS] DD0016: is the LinuxKit VM running?
[PASS] DD0011: are the LinuxKit services running?
[PASS] DD0004: is the Docker engine running?
[PASS] DD0015: are the binary symlinks installed?
[PASS] DD0031: does the Docker API work?
[PASS] DD0032: do Docker networks overlap with host IPs?

Please investigate the following 1 issue:

1 : The test: is the VM networking working?
    Failed with: network checks failed: failed to ping host: exit status 1

VM seems to have a network connectivity issue. Check your host firewall and anti-virus settings in case they are blocking the VM.
```

### Steps to reproduce the behavior

  1. `/Applications/Docker.app/Contents/MacOS/Docker --uninstall`

**Comment by nicks** [comment]
thanks for reporting! i can confirm. i tried as far back as 4.9.0 so it might have been broken for a while :cry:

**Comment by mat007** [comment]
This will be fixed in the next release of Docker Desktop (albeit with a slightly different command invocation).

**Comment by mat007** [comment]
Closing this issue because a fix has been released in Docker Desktop 4.16. See the [release notes](https://docs.docker.com/desktop/release-notes/) for more details.
